python/tvm/contrib/hexagon/session.py

```diff
@@ -93,7 +93,8 @@ def __enter__(self):
             raise exception
 
     def __exit__(self, exc_type, exc_value, exc_traceback):
-        pass
+        # close session to the tracker
+        del self._rpc
```

Comment:
Assuming that this is the only reference to `self._rpc`, this would destroy the object, because CPython uses a reference count in addition to generational garbage collection.  That said, that's more of an implementation detail, and is pretty fragile to rely upon.

It looks like the key function we want to call is [`RPCEndpoint::Shutdown`](https://github.com/apache/tvm/blob/main/src/runtime/rpc/rpc_endpoint.cc#L708), usually called when the `RPCEndpoint` object is destructed.  Unfortunately, I don't see any method in either [`RPCClientSession`](https://github.com/apache/tvm/blob/main/src/runtime/rpc/rpc_endpoint.cc#L995) or [`RPCSession`](https://github.com/apache/tvm/blob/main/src/runtime/rpc/rpc_session.h#L48).  I think we could add a `close` method to the python `RPCSession` object, which would then be called from `__exit__`.

```python
def close(self):
    self._sess.__del__()
    self._sess = None
```